### PR TITLE
Feature- update appMetadata

### PR DESCRIPTION
**Pull request by leightkt**
### Description
Allow for the updating of appMetadata after initialization

### Checklist
- [x] Increment the version field in `package.json` of the package you have made changes in following [semantic versioning](https://semver.org/) and using alpha release tagging
- [x] Check the box that allows repo maintainers to update this PR
- [x] Test locally to make sure this feature/fix works
- [x] Run `yarn check-all` to confirm there are not any associated errors
- [x] Confirm this PR passes Circle CI checks
- [x] Add or update relevant information in the documentation

### Docs Checklist
- [x] Include a screenshot of any changes ([see docs README on running locally](https://github.com/blocknative/web3-onboard/blob/develop/docs/README.md))
- [x] Add/update the appropriate package README (if applicable)
- [NA] Add/update the related module in the [docs demo](https://github.com/blocknative/web3-onboard/blob/develop/docs/src/lib/services/onboard.js) (if applicable)
- [x] Add/update the related package in the `docs/package.json` file (if applicable)

Screenshot 2023-06-14 at 8 30 45 PM

### CODE REVIEW

File: packages/core/src/index.ts [hunk 1]
```diff
@@ -217,6 +218,8 @@ function init(options: InitOptions): OnboardAPI {
     unstoppableResolution
   })
 
+  updateAppMetadata(appMetadata)
```
Comment (Adamj1232) [inline packages/core/src/index.ts:221]: I think we need an undefined check here and remove the default null initializer on ln 95

Comment (leightkt) [reply packages/core/src/index.ts]: updated!

File: packages/core/src/index.ts [hunk 2]
```diff
@@ -217,6 +218,8 @@ function init(options: InitOptions): OnboardAPI {
     unstoppableResolution
   })
 
+  appMetadata ? updateAppMetadata(appMetadata) : null
```
Comment (Adamj1232) [inline packages/core/src/index.ts:221]:
```suggestion
  appMetadata && updateAppMetadata(appMetadata)
```

Comment (Adamj1232) [reply packages/core/src/index.ts:221]: This is a little more clean than a ternary that resolves to null. Resolving to null would make more sense if the ternary was used to set a variable.
A basic if block would be fine also

File: packages/demo/src/App.svelte [hunk 3]
```diff
@@ -307,23 +307,23 @@
...
-      ],
-      agreement: {
-        version: '1.0.0',
-        termsUrl: 'https://www.blocknative.com/terms-conditions',
-        privacyUrl: 'https://www.blocknative.com/privacy-policy'
-      },
-      gettingStartedGuide: 'https://blocknative.com',
-      explore: 'https://blocknative.com'
-    },
+    // appMetadata: {
```
Comment (Adamj1232) [inline packages/demo/src/App.svelte:310]: This can be commented back in

**Review by Adamj1232 — APPROVED** [review]
Nice!!

**Pull request merged**